From fc658e3113bde26e72dc29d33567756331a05116 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 9 May 2019 00:15:28 +0200
Subject: [PATCH 1/1] efi_loader: EFI_DEVICE_PATH_UTILITIES_PROTOCOL

The EFI_DEVICE_PATH_UTILITIES_PROTOCOL is not needed for EBBR compliance.
So let's make it a customizable option.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/Kconfig         | 7 +++++++
 lib/efi_loader/Makefile        | 2 +-
 lib/efi_loader/efi_root_node.c | 2 ++
 lib/efi_selftest/Makefile      | 3 ++-
 4 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/lib/efi_loader/Kconfig b/lib/efi_loader/Kconfig
index 0307d731a8..4f79a9fcf4 100644
--- a/lib/efi_loader/Kconfig
+++ b/lib/efi_loader/Kconfig
@@ -18,6 +18,13 @@ config EFI_LOADER
 
 if EFI_LOADER
 
+config EFI_DEVICE_PATH_UTILITIES_PROTOCOL
+	bool "Provice device path utilities protocol"
+	default y
+	help
+	  The device path utilities protocol is used to create and manipulate
+	  device paths and their nodes. It is required to run the UEFI Shell.
+
 config EFI_LOADER_HII
 	bool "Expose HII protocols to EFI applications"
 	default y
diff --git a/lib/efi_loader/Makefile b/lib/efi_loader/Makefile
index 2b1ae61a3d..9576bad0da 100644
--- a/lib/efi_loader/Makefile
+++ b/lib/efi_loader/Makefile
@@ -22,7 +22,7 @@ obj-y += efi_boottime.o
 obj-y += efi_console.o
 obj-y += efi_device_path.o
 obj-y += efi_device_path_to_text.o
-obj-y += efi_device_path_utilities.o
+obj-$(CONGIG_EFI_DEVICE_PATH_UTILITIES_PROTOCOL) += efi_device_path_utilities.o
 obj-y += efi_file.o
 obj-$(CONFIG_EFI_LOADER_HII) += efi_hii.o efi_hii_config.o
 obj-y += efi_image_loader.o
diff --git a/lib/efi_loader/efi_root_node.c b/lib/efi_loader/efi_root_node.c
index f36ca3456e..9ce76edfd4 100644
--- a/lib/efi_loader/efi_root_node.c
+++ b/lib/efi_loader/efi_root_node.c
@@ -55,9 +55,11 @@ efi_status_t efi_root_node_register(void)
 			 /* Device path to text protocol */
 			 &efi_guid_device_path_to_text_protocol,
 			 (void *)&efi_device_path_to_text,
+#if CONFIG_IS_ENABLED(EFI_DEVICE_PATH_UTILITIES_PROTOCOL)
 			 /* Device path utilities protocol */
 			 &efi_guid_device_path_utilities_protocol,
 			 (void *)&efi_device_path_utilities,
+#endif
 #if CONFIG_IS_ENABLED(EFI_UNICODE_COLLATION_PROTOCOL)
 			 /* Unicode collation protocol */
 			 &efi_guid_unicode_collation_protocol,
diff --git a/lib/efi_selftest/Makefile b/lib/efi_selftest/Makefile
index 7fdf189c5c..3e420f6a8f 100644
--- a/lib/efi_selftest/Makefile
+++ b/lib/efi_selftest/Makefile
@@ -18,7 +18,6 @@ efi_selftest_controllers.o \
 efi_selftest_console.o \
 efi_selftest_crc32.o \
 efi_selftest_devicepath.o \
-efi_selftest_devicepath_util.o \
 efi_selftest_events.o \
 efi_selftest_event_groups.o \
 efi_selftest_exception.o \
@@ -38,6 +37,8 @@ efi_selftest_util.o \
 efi_selftest_variables.o \
 efi_selftest_watchdog.o
 
+obj-$(CONGIG_EFI_DEVICE_PATH_UTILITIES_PROTOCOL) += \
+efi_selftest_devicepath_util.o
 obj-$(CONFIG_EFI_UNICODE_COLLATION_PROTOCOL) += efi_selftest_unicode_collation.o
 
 obj-$(CONFIG_CPU_V7) += efi_selftest_unaligned.o
-- 
2.20.1

